src/vs/workbench/api/browser/mainThreadTask.ts

```diff
@@ -433,7 +433,9 @@ export class MainThreadTask implements MainThreadTaskShape {
 				let resolvedDefinition: ITaskDefinitionDTO = execution.task!.definition;
 				if (execution.task?.execution && CustomExecutionDTO.is(execution.task.execution) && event.resolvedVariables) {
 					const dictionary: IStringDictionary<string> = {};
-					Array.from(event.resolvedVariables.entries()).forEach(entry => dictionary[entry[0]] = entry[1]);
+					for (const entry of Object.entries(event.resolvedVariables)) {
```

Comment:
We should do this instead of `Object.entries` if it works:

```suggestion
					for (const entry of event.resolvedVariables.entries()) {
```